high_school/views.py

```diff
@@ -99,19 +100,16 @@ def get_context_data(self, *, object_list=None, **kwargs):
             context["empty_list"] = False
         else:
             context["unauth"] = False
-            if self.dbn:
-                context["selected_school"] = get_object_or_404(HighSchool, dbn=self.dbn)
+            if not context["high_schools"]:
+                context["empty_list"] = True
             else:
-                context["selected_school"] = None
-
-            high_schools = self.getHighSchools()
-            if high_schools:
-                context["high_schools"] = high_schools
+                if self.dbn:
+                    context["selected_school"] = get_object_or_404(
```

Comment:
I don't think get_object_or_404 should be used. This would just show an error page in case the school is not found. Instead we should handle this gracefully and show a page which says school does not exist.

Reply:
Agreed. Handling this now.